nodejs/dist/uws.js

```diff
@@ -209,9 +210,16 @@ class Server extends EventEmitter {
                 options.path = '/' + options.path;
             }
 
-            this.httpServer.on('upgrade', (request, socket, head) => {
-                if (!options.path || options.path == request.url.split('?')[0].split('#')[0]) {
-                    if (options.verifyClient) {
+            this._onServerUpgrade = (request, socket, head) => {
+
+                // check for wrong path
+                if (options.path) {
+                  var u = url.parse(request.url);
+                  if (u && u.pathname !== options.path)
+                    return;
```

Comment:
Hahaha :smile: no idea such flame was firing here 

As it's not settled (and my guess is that it will not be soon) feel free to align with ws behavior (or not, it's your call !) 

I'll stick in my fork until them :cake:. (make sure to cleanup the handle, though)